### ISSUE TITLE
error compiling development version...

### ISSUE THREAD

**Issue by seannaswell** [zotonic/zotonic#1938]
Using Erlang v19, getting the following error when compiling the development version of Zotonic:

> ===>Compiling /home/zotonic/zotonic/_build/default/lib/erlexec/c_src/exec.cpp                              
===> /home/zotonic/zotonic/_build/default/lib/erlexec/c_src/exec.cpp:474:14: error: #error setresuid(3) not supported!
             #error setresuid(3) not supported!
              ^
/home/zotonic/zotonic/_build/default/lib/erlexec/c_src/exec.cpp: In function 'void initialize(int, bool, bool)':
/home/zotonic/zotonic/_build/default/lib/erlexec/c_src/exec.cpp:476:9: error: expected primary-expression before '<' token
         < 0) {
         ^

**Comment by seannaswell** [comment]
Using 0.41 for now, just FYI. Thanks.

**Comment by mworrell** [comment]
Hi @seannaswell, thanks for your report.  On which OS are you compiling?

**Comment by seannaswell** [comment]
Compiled both development and 0.41 versions on DragonFlyBSD.

Just FYI: when compiling 0.41 I do have to modify the Proper rebar.config file, as reflected in[ this commit](https://github.com/proper-testing/proper/commit/ad8d80f80cba6c126b5b37facf7bb1ac9950569e). However, once that is done 0.41 compiles and functions as expected.

The error referenced does not appear to be related to the Proper rebar.config file, which is the reason for the report.

**Comment by mworrell** [comment]
Ah, I guess we need to update proper then, I made a separate issue.  (Issue #1942)

**Comment by mworrell** [comment]
For `erlexec` we need to look into some BSD specifics.

@CyBeRoni do you have any ideas for BSD (as you are running that as well)

**Comment by CyBeRoni** [comment]
I have actually never run BSD, so I'm afraid not.

**Comment by mworrell** [comment]
That is a pity - do you know somebody who still does?

**Comment by mworrell** [comment]
BTW @CyBeRoni do you still run Solaris?  If so, can you check if you can compile master?

**Comment by seannaswell** [comment]
If there is anything I can do to help, please let me know.

**Comment by mworrell** [comment]
We have to make up our mind what to do with erlexec on BSD (and Windows).

There are two options:

  1. Fix erlexec to work on those operating systems (even if it is with a subset)
  2. Use something else on those operation systems

We use erlexec as it gives us much better control over the running processes than `os:cmd/1` (and friends) could give us.

**Comment by seannaswell** [comment]
Won't comment on Windows, but all of the BSD's, MacOS and Solaris all come from the [same source](https://en.wikipedia.org/wiki/BSD), and according to [this page](https://github.com/saleyn/erlexec), FreeBSD is supported.

Will look into erlexec more closely, thanks for the clarification.

**Comment by seannaswell** [comment]
Very interesting search result turned up when looking for FreeBSD and setresuid, someone commented about building Zotonic on FreeBSD: https://forums.freebsd.org/threads/setresuid-3.64944/

Has there been a regression in erlexec, or is this a new issue?

**Comment by mworrell** [comment]
That is interesting, now that I see that discussion I remember handling it.
Let's check the version of erlexec we are using.

**Comment by mworrell** [comment]
I have upgraded erlexec to 1.9.3.

Does rebar3 classify your system as `freebsd`?

https://github.com/saleyn/erlexec/commit/7a00673044a3037310cb3a15ff829a879f41d6a8#diff-f310c7cc098862c9835f2ba9675a4060

**Comment by seannaswell** [comment]
That is likely the issue, it probably reports as “dragonfly”, as it does with Proper. Will test.

Sent with my thumbs

**Comment by seannaswell** [comment]
Ok, it appears that with a few minor changes, 1.0 development now compiles on DragonFly: https://github.com/saleyn/erlexec/compare/master...seannaswell:master

DragonFlyBSD is a fork of FreeBSD therefore erlexec should work the same on both, however, at this point I can only say for certain that it compiles. 

Perhaps you can point to functionality that Zotonic uses erlexec for that will break if erlexec is not working properly, so I can verify that it indeed does work. Alternatively, if you suggest another approach, please let me know.

**Comment by mworrell** [comment]
One of the primary places we use it is in the filewatcher, to keep track of changes using fswatch or inotify,

**Comment by mworrell** [comment]
I see that it is already merged.
Ping me when you see a new release appearing (as we use Hex.pm for erlexec).

**Comment by seannaswell** [comment]
Ok, will do.
